airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -403,6 +403,37 @@ def stream_slices(self) -> Iterable[StreamSlice]:
                 }
             )
 
+class BulkParentStreamStreamSlicer(StreamSlicer):
+    def __init__(self, batched_substream, sync_mode: SyncMode, cursor_field: Optional[List[str]], stream_state: Optional[Mapping[str, Any]], parend_id_field: str) -> None:
```

Comment:
what's a batched_substream? Can you type it?

Reply:
It is a stream that inherits from BatchedSubStream and hopefully does not re-implement `stream_slices`. I'll add the typing